File: app/src/main/java/nural/smart/cdleganes/EntryDetailActivity.java
package nural.smart.cdleganes;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.view.MenuItemCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.ShareActionProvider;
import android.view.Menu;
import android.view.MenuItem;
import android.webkit.WebView;
import android.widget.ProgressBar;

import com.google.firebase.analytics.FirebaseAnalytics;

import nural.smart.cdleganes.components.MyWebViewClient;

public class EntryDetailActivity extends AppCompatActivity {

    private FirebaseAnalytics mFirebaseAnalytics;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_entry_detail);

        mFirebaseAnalytics = FirebaseAnalytics.getInstance(this);

        String title = this.getIntent().getExtras().getString("title");
        String url = this.getIntent().getExtras().getString("url");

        setTitle(title);

        final ProgressBar progressBar = (ProgressBar) findViewById(R.id.progressBarDetail);
        final WebView mWebView = (WebView) findViewById(R.id.detail_web_view);
        mWebView.setWebViewClient(new MyWebViewClient(progressBar));

        mWebView.loadUrl(url);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate menu resource file.
        getMenuInflater().inflate(R.menu.menu_main, menu);

        // Locate MenuItem with ShareActionProvider
        MenuItem item = menu.findItem(R.id.menu_item_share);

        // Fetch and store ShareActionProviderm
        final ShareActionProvider mShareActionProvider = (ShareActionProvider) MenuItemCompat.getActionProvider(item);

        // Return true to display menu
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle presses on the action bar items
        switch (item.getItemId()) {
            case R.id.menu_item_share:
                String url = this.getIntent().getExtras().getString("url");

                Intent sendIntent = new Intent();
                sendIntent.setAction(Intent.ACTION_SEND);
                sendIntent.putExtra(Intent.EXTRA_TEXT, url +  " " + "App: Noticias Leganes, Twitter: @ThisIsButarque");
                sendIntent.setType("text/plain");
                startActivity(Intent.createChooser(sendIntent, "Compartir"));

                Bundle bundle = new Bundle();
                bundle.putString(FirebaseAnalytics.Param.ITEM_ID, url);
                mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.SHARE, bundle);

                return true;

            default:
                return super.onOptionsItemSelected(item);
        }
    }


}

File: app/src/main/java/nural/smart/cdleganes/EntryDetailActivity.kt
package nural.smart.cdleganes

/**
 * Created by alvaro on 23/11/17.
 */


import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso


class EntryDetailActivity: AppCompatActivity() {
    companion object {
        val TITLE = "title"
        val DATE = "date"
        val DESCRIPTION = "description"
        val IMAGE = "image"
        val URL = "url"
        val ORIGEN = "origen"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_entry_detail)

        val titleTextView = findViewById(R.id.titleEntryDetail) as TextView
        val dateTextView = findViewById(R.id.dateEntryDetail) as TextView
        val descriptionTextView = findViewById(R.id.descriptionEntryDetail) as TextView
        val moreButton = findViewById(R.id.buttonEntryDetail) as Button
        val imageView = findViewById(R.id.imageEntryDetail) as ImageView

        titleTextView.text = intent.getStringExtra(TITLE)
        dateTextView.text = intent.getStringExtra(DATE)
        var description = intent.getStringExtra(DESCRIPTION)
        if (description.length > 0 && description.substring(description.length - 1) != ".") {
            description = description + "..."
        }
        descriptionTextView.text = description
        moreButton.text = "Ver noticia en " + intent.getStringExtra(ORIGEN)
        val imageUrl = intent.getStringExtra(IMAGE)

        Picasso.with(this).load(imageUrl).placeholder(R.mipmap.ic_launcher_bn).into(imageView)
        val url = intent.getStringExtra(URL)

        moreButton.setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            startActivity(intent)
        }
    }

}